internal/services/containers/container_registry_cache_rule_resource_test.go

```diff
@@ -0,0 +1,103 @@
+package containers_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/cacherules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ContainerRegistryCacheRuleResource struct{}
+
+func TestAccContainerRegistryCacheRule_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_cache_rule", "test")
+	r := ContainerRegistryCacheRuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccContainerRegistryCacheRule_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_cache_rule", "test")
+	r := ContainerRegistryCacheRuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_container_registry_cache_rule"),
+		},
+	})
+}
+
+func (t ContainerRegistryCacheRuleResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := cacherules.ParseCacheRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Containers.ContainerRegistryClient_v2023_07_01.CacheRules.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (ContainerRegistryCacheRuleResource) basic(data acceptance.TestData) string {
```

Comment:
We don't appear to have any test that sets `credential_set_id`, can we add this?